src/app/plugins/marketplace.plugin.ts

```diff
@@ -49,23 +49,31 @@ export class MarketPlacePlugin extends Plugin {
     this._replyToUser(message, embed);
   }
 
-  private async _handleListMarket(message: IMessage) {
-    const itemsForSale = await this._fetchMessages(message, 300).catch((e) =>
-      this.container.loggerService.error(e)
-    );
-
-    if (!itemsForSale) {
-      return;
-    }
+  private async _handleListMarket(message: IMessage, shouldDmUser: boolean) {
+    const oldMessages = await this._fetchMessages(message, 300);
+    const itemsForSale = await this._fetchListings(oldMessages);
+    this._deleteOldListingPosts(oldMessages);
 
     const embed = new RichEmbed();
     embed.setTitle('Items For Sale');
     embed.setColor('#7289da');
     embed.setDescription(itemsForSale.reverse().join('\n\n'));
-    this._replyToUser(message, embed);
+
+    message.channel.send(embed);
+    if (shouldDmUser) {
+      this._replyToUser(message, embed);
+    }
     message.delete();
   }
 
+  private _deleteOldListingPosts(messages: Message[]) {
```

Comment:
consider that only one listing post "should" exist above (any others can just be buried, it's fine)

why not keep `Message` object for the last non-DM Listings Message, then when you make a new one, delete the old one, then save the new one as the new "last" 

this way we dont fetch and loop, which is slow.

Reply:
True, I'm conditioned to try to commit as little to memory as possible so I didn't even think of that 😄